src/mainloop.cpp

```diff
@@ -217,14 +215,27 @@ void Mainloop::camera_callback(const void *img, size_t len, const struct timeval
 
 	Point3_<double> gyro_data;
 	struct timespec gyro_timespec;
-	_bmi->gyro_integrated_get(&gyro_data, &gyro_timespec);
+	static struct timespec _gyro_last_timespec = gyro_timespec;
+	static Point3_<double> gyro_integrated_data(0.0, 0.0, 0.0);
+
+	_bmi->get_gyro_sample(&gyro_data, &gyro_timespec);
```

Comment:
Huum but BMI160 output rate is 1600Hz the camera is 30Hz(due driver problems) so several gyro samples will just be discarded, this can give a non-precise value.
I can see that the gyro integrated is reset after send the message to PX4, so if you want I get work in integrate all the samples and reset it after send the message.